tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/tests/PitclipseRunConfigurationMutationTabTest.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    @Test
+    public void useStrongerMutatorsGroup() {
+        // now set STRONGER mutators
+        PAGES.getRunMenu().setMutatorGroup(TEST_CONFIG_NAME, Mutators.STRONGER);
+        // run test and confirm result is as expected
+        PAGES.getRunMenu().runPitWithConfiguration(TEST_CONFIG_NAME);
+        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 0);
+        mutationsAre(   "SURVIVED    | " + TEST_PROJECT + " | foo.bar | foo.bar.Foo |    7 | negated conditional\n" +
+                        "SURVIVED    | " + TEST_PROJECT + " | foo.bar | foo.bar.Foo |    7 | removed conditional - replaced equality check with false\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Bar |    7 | negated conditional\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Bar |    7 | removed conditional - replaced equality check with false\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Bar |    8 | Replaced integer addition with subtraction\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Bar |    8 | replaced int return with 0 for foo/bar/Bar::f\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Foo |    8 | Replaced integer addition with subtraction\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Foo |    8 | replaced int return with 0 for foo/bar/Foo::f");
+    }
+
+    @Test
+    public void checkOneMutant() {
+        PAGES.getRunMenu().setOneCustomMutator(TEST_CONFIG_NAME, Mutators.NEGATE_CONDITIONALS);
+        // run test and confirm result is as expected
+        PAGES.getRunMenu().runPitWithConfiguration(TEST_CONFIG_NAME);
+        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 0);
+        // check that mutator was selected as only mutator
+        mutatorIs(Mutators.NEGATE_CONDITIONALS);
+        mutationsAre(   "SURVIVED    | " + TEST_PROJECT + " | foo.bar | foo.bar.Foo |    7 | negated conditional\n" +
+                        "NO_COVERAGE | " + TEST_PROJECT + " | foo.bar | foo.bar.Bar |    7 | negated conditional");
+    }
+
+    @Test
+    public void useAllMutatorsGroup() {
+        // now set ALL mutators group
+        PAGES.getRunMenu().setMutatorGroup(TEST_CONFIG_NAME, Mutators.ALL);
+        // run test and confirm result is as expected
+        PAGES.getRunMenu().runPitWithConfiguration(TEST_CONFIG_NAME);
+        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 1);
+        mutationsAre(getAllMutantsResult());
+    }
+    
+    @Test
+    public void checkAllMutants() {
+        PAGES.getRunMenu().checkAllMutators(TEST_CONFIG_NAME);
+        // run test and confirm result is as expected
+        PAGES.getRunMenu().runPitWithConfiguration(TEST_CONFIG_NAME);
+        coverageReportGenerated(TESTED_CLASSES, COVERAGE, 1);
```

Comment:
For some reason my maven clean verify fails at this point with 0% MT coverage, but running it with eclipse it is 1%.
Also, the test above using the "All" group works fine.

Reply:
Was this the reason why the `PitclipseOptionsTest` is not testing this case or because of newly added mutators would break the test?

Reply:
Maybe that's related to the above hint on making each single test method executable independently from possible side effects of other methods.

Reply:
I'm not sure what you are saying. Would you mind explaining again? Or what hint are you referring to?

Reply:
Maybe I did not get the tests right and so my comment was confusing... I was wondering if customizing a run configuration might change some general Pitclipse preferences that might have side effects on further tests... if that's not the case, then forget about my comment. If that does not solve your problem, then again I guess it's some synchronization problem and the check finds the previous output of PIT from a previous test? Are you sure that when `coverageReportGenerated` is executed PIT has finished executing?

Reply:
The preferences are untouched. Only the run configurations are changed.

I now can confirm that it is indeed a problem with synchronization. It goes back to the same issue I already fixed in #150.

The way the synchronization works is not really sufficient to work for longer pit runs.

@LorenzoBettini @echebbi I would like to wait with this PR, until #150 is merged and fixes this issue. Would this be ok?

Reply:
Could you please be more specific on the part of test code where you fixed that?
If that's something that's useful in general for UI tests, maybe it's worthwhile to

- create a separate PR with the fix in the test code (again, to avoid having too many changes in a PR), or
- also use the same fix also in this PR, so that we're sure it actually fixes the problem (in that case you might get a merge conflict depending on which PR is merged first, but that should be easy to handle)

@echebbi what do you think?

Also because #150 still has test failures (I still haven't reviewed that)

Reply:
Moreover, please remember that the current synchronization is based on calling `PitNotifier.INSTANCE.reset();' before running PIT (see, e.g., `org.pitest.pitclipse.ui.behaviours.steps.PitclipseSteps.runPit(Runnable)`) so you must make sure to do that as well before running PIT

Reply:
This is done in the fix of #150, which we discussed there.

>
>Also because #150 still has test failures (I still haven't reviewed that)

@LorenzoBettini The tests should not fail anymore and I hope you could review it to approve the merge.

Reply:
https://github.com/pitest/pitclipse/pull/150#discussion_r671866331

@echebbi These are the comments I was talking about. It required deleting some files, using a different extension point and some more.

Reply:
so maybe it might be worthwhile to have a separate PR? @echebbi ?

Reply:
Yes, the changes seem significant enough to be put into a dedicated PR.

Reply:
@echebbi @LorenzoBettini Then please merge #167. It is the PR you requested.